### PR TITLE
Normalize decoders to Wazuh standard

### PR DESCRIPTION
**Pull request by pereyra-m**
|Related issue|
|---|
|#6773|

## Description

This PR:

- Updates the Wazuh code to print logs with **wazuh** instead of **ossec**. 
- Updates the name of some defines
- Updates the name of the Alert_Decoder plugin
- Creates the wazuh category to replace ossec

It also replicates the changes of the ruleset PR [797](https://github.com/wazuh/wazuh-ruleset/pull/797) into the **src** folder.

## Tests

- Compilation without warnings in every supported platform
  - [x] Linux
 - [x] Source installation
- [x] Review logs syntax and correct language

### CODE REVIEW

File: etc/rules/0015-ossec_rules.xml [hunk 1]
```diff
@@ -388,8 +388,8 @@
   </rule>
 
   <rule id="700" level="0">
-    <category>ossec</category>
-    <decoded_as>ossec-logcollector</decoded_as>
+    <category>wazuh</category>
+    <decoded_as>wazuh-logcollector</decoded_as>
```
Comment (DProvinciani) [inline etc/rules/0015-ossec_rules.xml:392]: Same question as in the ruleset repository. This already changed?

**Review by DProvinciani — CHANGES_REQUESTED** [review]
Just one comment to review and discuss.

File: src/analysisd/decoders/plugins/wazuhalert_decoder.c [hunk 2]
```diff
@@ -40,11 +40,12 @@ void *OSSECAlert_Decoder_Exec(Eventinfo *lf, OSHash *rules_hash, __attribute__((
     char *tmp_str = NULL;
     RuleInfo *rule_pointer;
 
-    lf->decoder_info->type = OSSEC_ALERT;
+    lf->decoder_info->type = WAZUH_ALERT;
 
     /* Checking the alert level. */
     if (strncmp("Alert Level: ", lf->log, 12) != 0 &&
-       strncmp("ossec: Alert Level:", lf->log, 18) != 0) {
+        strncmp("ossec: Alert Level:", lf->log, 18) != 0 &&
```
Comment (palaciosjeremias) [inline src/analysisd/decoders/plugins/wazuhalert_decoder.c:47]: Same observation as in ruleset PR: I think these logs are only part of the manager and don't require this dual comparison.

File: etc/rules/0015-ossec_rules.xml (hunk 1 above)
Comment (pereyra-m) [reply etc/rules/0015-ossec_rules.xml:392]: The _wazuh-logcollector_ decoder was already added, replacing _ossec-logcollector_.

File: src/analysisd/decoders/plugins/wazuhalert_decoder.c (hunk 2 above)
Comment (pereyra-m) [reply src/analysisd/decoders/plugins/wazuhalert_decoder.c:47]: Updated!

**Review by palaciosjeremias — APPROVED** [review]
LGTM

**Review by DProvinciani — APPROVED** [review]
Looks good!

**Pull request merged**